### ISSUE TITLE
flaky extension test

### ISSUE THREAD

**Issue by josharian** [cursorless-dev/cursorless#1730]
This test flakes on my machine about half the time: `recorded/actions/pourLine:`.

The failure mode is 100% consistent, having to do with indentation:

```
 1) recorded test cases
       recorded/actions/pourLine:

      Unexpected final state
      + expected - actual

       {
      -  "documentContents": "class MyClass:\n    def my_funk():\n    \n        pass\n"
      +  "documentContents": "class MyClass:\n    def my_funk():\n        \n        pass\n"
         "selections": [
           {
             "active": {
      -        "character": 4
      +        "character": 8
               "line": 2
             }
             "anchor": {
      -        "character": 4
      +        "character": 8
               "line": 2
             }
           }
         ]
 ```

Any suggestions for how I can go about stabilizing this test?  I wonder what it is about my machine that  It causes it to fail just for me. Maybe something about my python configuration  Or other VS Code settings?

**Comment by pokey** [comment]
Should be able to increase the sleep in that test a little bit. See the fixture and search for `sleep`

**Comment by josharian** [comment]
Tried doubling it. And adding `postCommandSleepTimeMs: 200` as well. No dice.

My python settings are:

```
  "[python]": {
    "editor.formatOnType": false,
    "editor.formatOnSave": true,
    "editor.defaultFormatter": "ms-python.black-formatter"
  },
```

Seems ok.

**Comment by pokey** [comment]
Huh. So the issue is that it should be relying on the VSCode Python extension to perform indentation there. My theory was that it just wasn't giving Python extension time to start up. Maybe try making the timeout stupidly long just to be sure that's not the issue? Note that it will retry with successively longer sleeps (I think it doubles 3 times)

**Comment by nathanheffley** [comment]
I have this same issue, even worse though. It almost never passes locally. If I bump the `postEditorOpenSleepTimeMs` to `5000` it passes occassionally.
Is the indentation important enough to this test that if I changed this test to a `plaintext` test like other `pour` tests, it would be detrimental to the test suite?

**Comment by pokey** [comment]
Unfortunately, yes, the sole purpose of this test is to check that we let the language server determine indentation when we do a pour line. I wonder what else we could try 🤔

I guess worst case we could have a flag to disable it locally, but adding that flag feels like opening a can of worms we don't want to open

We could potentially try adding a sleep in the action itself that only runs during test?

**Comment by nathanheffley** [comment]
Figured that may be the case. Using the test subset feature makes it pretty easy to work around anyway. More complexity certainly seems unnecessary for one test.

**Comment by pokey** [comment]
yeah or just ignoring that failure for now 🤷. but yeah if you only care about php tests then not a big deal to just use subset
